src/module_2/module_2_data.md

```diff
@@ -0,0 +1,1750 @@
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>user_id</th>
+      <th>variant_id</th>
+      <th>created_at</th>
+      <th>price</th>
+      <th>compare_at_price</th>
+      <th>vendor</th>
+      <th>product_type</th>
+      <th>tags</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>68e872ff888303bff58ec56a3a986f77ddebdbe5c279e7...</td>
+      <td>33618848088196</td>
+      <td>2020-04-30 15:07:03</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>aed88fc0b004270a62ff1fe4b94141f6b1db1496dbb0c0...</td>
+      <td>33667178659972</td>
+      <td>2020-05-05 23:34:35</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>68e872ff888303bff58ec56a3a986f77ddebdbe5c279e7...</td>
+      <td>33619009208452</td>
+      <td>2020-04-30 15:07:03</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>aed88fc0b004270a62ff1fe4b94141f6b1db1496dbb0c0...</td>
+      <td>33667305373828</td>
+      <td>2020-05-05 23:34:35</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>4594e99557113d5a1c5b59bf31b8704aafe5c7bd180b32...</td>
+      <td>33667247341700</td>
+      <td>2020-05-06 14:42:11</td>
+      <td>3.49</td>
+      <td>3.5</td>
+      <td>method</td>
+      <td>cleaning-products</td>
+      <td>[cruelty-free, eco, vegan, window-glass-cleaner]</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+df_regulars_inventory.dropna(subset=['price'], inplace=True)
+df_regulars_inventory.info()
+```
+
+    <class 'pandas.core.frame.DataFrame'>
+    Index: 15034 entries, 4 to 18104
+    Data columns (total 8 columns):
+     #   Column            Non-Null Count  Dtype         
+    ---  ------            --------------  -----         
+     0   user_id           15034 non-null  object        
+     1   variant_id        15034 non-null  int64         
+     2   created_at        15034 non-null  datetime64[us]
+     3   price             15034 non-null  float64       
+     4   compare_at_price  15034 non-null  float64       
+     5   vendor            15034 non-null  object        
+     6   product_type      15034 non-null  object        
+     7   tags              15034 non-null  object        
+    dtypes: datetime64[us](1), float64(2), int64(1), object(4)
+    memory usage: 1.0+ MB
+
+
+
+```python
+#Checking number of regular products bought by user
```

Comment:
If I understand this well, this is not what the plot is showing. Instead is showing something like: "Number of users having a concrete product as regular, for the most popular products"

Reply:
Yes, it's true. To accomplish the stated in the comment would be the other way around right?

From
```
#Checking number of regular products bought by user
num_products = df_regulars_inventory.groupby('variant_id')['user_id'].nunique().reset_index().sort_values('user_id', ascending=False).head(90)
```
to
```
#Checking number of regular products bought by user
num_products = df_regulars_inventory.groupby('user_id')['variant_id'].count().reset_index().sort_values('variant_id', ascending=False).head(90)
```